hathor/transaction/poa/poa_block.py

```diff
@@ -73,8 +79,23 @@ def get_graph_fields_from_struct(self, buf: bytes, *, verbose: VerboseCallback =
 
         return buf
 
+    @override
     def get_graph_struct(self) -> bytes:
         assert len(self.signer_id) == poa.SIGNER_ID_LEN
         struct_bytes_without_poa = super().get_graph_struct()
         signature_len = int_to_bytes(len(self.signature), 1)
         return struct_bytes_without_poa + self.signer_id + signature_len + self.signature
+
+    @override
+    def to_json(self, decode_script: bool = False, include_metadata: bool = False) -> dict[str, Any]:
+        poa_settings = self._settings.CONSENSUS_ALGORITHM
+        assert isinstance(poa_settings, PoaSettings)
+        json = super().to_json(decode_script=decode_script, include_metadata=include_metadata)
+        index_and_key = poa.get_signer_index_and_public_key(poa_settings, self.signer_id)
+
+        if index_and_key is not None:
+            _, signer = index_and_key
```

Comment:
You might return the wrong information if there are more than one match.

Reply:
This has been removed. See https://github.com/HathorNetwork/hathor-core/pull/1071#discussion_r1669469069

Reply:
Ok. Thanks!